### DISCUSSION TITLE
Diff Execution Environments

### DISCUSSION THREAD

**Discussion by my0373** [ansible/ansible-navigator#1234] [Ideas]
Hi, I'm wondering if there is any appetite to add a "diff" functionality between 2 execution environments?

A simple use case might be that if a user extends an existing EE but has to rebuild as the EE is upgraded, they can quickly spot any major changes that might break further down the line. 

It would hopefully make developers life easier as they don't need to document EE changes manually in a release note or similar.

I wasn't sure if this was the right place or potentially to be discussed here 
https://github.com/ansible/awx/blob/devel/docs/execution_environments.md

Would this be a worthwhile contribution to the project?

Thanks

Matt

**Comment by cidrblock** [comment]
I like the idea.  

Did you try getting the image information via stdout and dumping it to a file?  This could be done twice, once for each EE and those files could be diff'ed.

I'm not sure how useful the diff would be if the 2 EEs were _really_ different though...  

Let me know if that works, we can also talk about where and how to implement a smarter diff in navigator too

**Reply by my0373** [reply]
Awesome, thanks. I've not done much in terms of looking at code until I had bounced it off some others first. I will have a play over the next few evenings and see if I can develop a sensible approach. 

Thanks for taking the time to reply!
